sycl/doc/extensions/FilterSelector/FilterSelector.adoc

```diff
@@ -0,0 +1,85 @@
+= SYCL Proposals: Filter Selector
+James Brodman <james.brodman@intel.com>
+v0.1
+:source-highlighter: pygments
+:icons: font
+== Introduction
+This document presents an extension on top of the SYCL specification.  The goal of this extension is to provide a new device selector class that allows expressing common but non-trivial requirements for device selection in a simple manner.
+
+== Filter Selector
+
+The filter selector is a new device selector class that accepts a string of one or more filters that refine the set of devices that may be returned when the selector's `select_device` method is invoked.  Devices that match the specified filter(s) are ranked by the default_selector to determine which device is ultimately selected.
+
+=== DSL for Specifying Filters
+
+A string passed to the selector defines one or more filters.  Filters have a certain syntax that must be followed.  A filter is specified as a triple of the form:
```

Comment:
Do we define what happens if that syntax is not followed?

Reply:
Throws an error.

Reply:
> Throws an error.

Just to expand, syntax being incorrect throws and error, but what about requesting a GPU when there isn't one or requesting gpu:1 when only gpu:0 exists?

Reply:
You should get a runtime error (some sorta device not found one) similar to using gpu_selector when there's no gpu.